app/models/miq_queue.rb

```diff
@@ -486,6 +486,14 @@ def deliver(requester = nil, &block)
     return status, message, result
   end
 
+  # @return true if completed, false if timed out
+  def deliver_and_process(requester = nil, &block)
+    status, message, result = deliver(requester = nil, &block)
```

Comment:
Bug?  This will set requester to nil and completely kill any incoming parameter.  Do any of the callers even pass a requester?